File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int open = 0, close = 0;
        for(auto it : s){
            if(it == '('){
                open++;
            }
            else {
                if(open > 0){
                    open--;
                }else {
                    close++;
                }
            }
        }
        return open + close;
    }
};

File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        o,cl = 0,0

        for c in s:
            if c == '(':
                o+=1
            else:
                if o > 0:
                    o -=1
                else:
                    cl += 1
        return o + cl
        